File: javascript/backend-validator/07-multiple-subdivisions.mjs
// Set your CountriesDB private API key here
const API_KEY = 'your_private_api_key_here';

import { CountriesDBValidator } from '@countriesdb/validator';

if (!API_KEY || API_KEY === 'your_private_api_key_here') {
  throw new Error('Please set your COUNTRIESDB_PRIVATE_KEY in the script');
}

const validator = new CountriesDBValidator({
  apiKey: API_KEY,
});

// 7. Multiple subdivision validation
async function main() {
  try {
    console.log('Validating multiple subdivision codes: US-CA, US-NY, US-TX (country: US)');
    const results = await validator.validateSubdivisions(
      ['US-CA', 'US-NY', 'US-TX'],
      'US'
    );
    console.log('Response:', JSON.stringify(results, null, 2));
    
    if (results.some(r => !r.valid)) {
      throw new Error('Some subdivisions are invalid');
    }
    
    console.log('All subdivisions are valid');
  } catch (err) {
    console.error('Validation failed:', err.message);
    process.exit(1);
  }
}

main();

File: python/backend-validator/07-multiple-subdivisions.py
"""Multiple subdivision validation example using countriesdb-validator package"""

from validator import CountriesDBValidator
import json
import sys

# Set your CountriesDB private API key here
API_KEY = 'your_private_api_key_here'

if not API_KEY or API_KEY == 'your_private_api_key_here':
    raise SystemExit('Please set your API_KEY in the script')

validator = CountriesDBValidator({
    'api_key': API_KEY,
})

# 7. Multiple subdivision validation
def main():
    try:
        codes = ['US-CA', 'US-NY', 'US-TX']
        print(f'Validating subdivision codes: {codes} for country: US')
        results = validator.validate_subdivisions(codes, 'US')
        print('Response:', json.dumps(results, indent=2))
        
        for result in results:
            if not result['valid']:
                raise ValueError(f"Subdivision {result.get('code')} validation failed: {result.get('message')}")
        
        print('All subdivisions are valid')
    except Exception as exc:
        print(f'Validation failed: {exc}', file=sys.stderr)
        sys.exit(1)


if __name__ == '__main__':
    main()


